examples/13_write_dynamic_configuration.py

```diff
@@ -0,0 +1,155 @@
+#!/usr/bin/env python
+import json
+import openpmd_api as io
+import numpy as np
+
+
+def hasAdios2():
+    return 'adios2' in io.variants and io.variants['adios2']
+
+
+# This example demonstrates how to use JSON/TOML-based dynamic
+# configuration for openPMD.
+# The following configuration is passed to the constructor of the Series
+# class and specifies the defaults to used for that Series.
+# This configuration can later be overridden as needed on a per-dataset
+# level.
+
+defaults = """
+# This configuration is TOML-based
+# JSON can be used alternatively, the openPMD-api will automatically detect
+# the language being used
+#
+# Alternatively, the location of a JSON/TOML-file on the filesystem can
+# be passed by adding an at-sign `@` in front of the path
+# The format will then be recognized by filename extension, i.e. .json or .toml
+
+backend = "adios2"
+iteration_encoding = "group_based"
+# The following is only relevant in read mode
+defer_iteration_parsing = true
+
+[adios1.dataset]
+transform = "blosc:compressor=zlib,shuffle=bit,lvl=5;nometa"
+
+[adios2.engine]
+type = "bp4"
+
+# ADIOS2 allows adding several operators
+# Lists are given in TOML by using double brackets
+[[adios2.dataset.operators]]
+type = "zlib"
+
+parameters.clevel = 5
+# Alternatively:
+# [adios2.dataset.operators.parameters]
+# clevel = 9
+
+# For adding a further parameter:
+# [[adios2.dataset.operators]]
+# type = "some other parameter"
+# # ...
+
+[hdf5.dataset]
+chunks = "auto"
+"""
+
+
+def main():
+    if not hasAdios2():
```

Comment:
We intentionally always contain all known keys, so you can simplify to:
```suggestion
    if not io.variants['adios2']:
```
if you like.